src/cascadia/TerminalConnection/ConhostConnection.h

```diff
@@ -26,20 +26,22 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         winrt::event<TerminalConnection::TerminalOutputEventArgs> _outputHandlers;
         winrt::event<TerminalConnection::TerminalDisconnectedEventArgs> _disconnectHandlers;
 
-        uint32_t _initialRows{};
-        uint32_t _initialCols{};
-        hstring _commandline{};
-        hstring _startingDirectory{};
-
-        bool _connected{};
-        HANDLE _inPipe{ INVALID_HANDLE_VALUE };  // The pipe for writing input to
-        HANDLE _outPipe{ INVALID_HANDLE_VALUE }; // The pipe for reading output from
-        HANDLE _signalPipe{ INVALID_HANDLE_VALUE };
-        DWORD _outputThreadId{};
-        HANDLE _hOutputThread{ INVALID_HANDLE_VALUE };
-        PROCESS_INFORMATION _piConhost{};
-        guid _guid{}; // A "unique" session identifier for connected client
-        bool _closing{};
+        // set during construction - not constant

```

Comment:
Even though these are initialized in the constructor, there's no harm in also initializing them here.  The compiler is smart enough to do the right thing.

Personally, I prefer to initialize them in both places because it ensures that members of the class are never in an invalid or unknown state.  For example, if the members are initialized in the class constructor, it ensures they have a sane default value until they are explicitly initialized in the constructor.  In addition, it also means that if someone removes the initialization from the constructor later, there will still be a good default value.

With that said, if this is what the project prefers, that's fine.

Reply:
I agree with @binarycrusader.   This "rule"  - makes it easier for the developer to get the original guideline right and it makes the code review much simpler.

Reply:
So here's a question: for things that have a meaningful default constructor, do we still put `{}` in the class? For POD types it's obviously required, but for `winrt::hstring` and `wil::unique_x` it's implicit and it's always initialized properly. Thoughts?

Reply:
@DHowett-MSFT In my day job, we don't add {} for classes/objects.  
Our rule has more detail with proper "c++ legalese"  but it is basicallly
Use {} for all POD,  structs.  (i.e. - any object that doesn't get initialized at declaration)
For enums and enum class we require that you provide a default enum value.  (enums with {} have the potential to be completely wrong)

Since structs can be default initialized correctly with member initializing you could argue that they don't need to use {}.   But again to keep this simple we require the use of {} because most structs do not member initialize their variables.

Not sure if this is right, but seeing objects that have ctors that have {} seems wrong to us.

Reply:
Note - you can have execptions like winrt::hstring.  It is a struct but you can treat it like an object because we are 100% sure it will be initialized correctly.  (I don't have the source on hand for winrt::string so I am asumming this based on your comments)

Reply:
Looking at winrt::hstring closer - if you were to have a class member that is an hstring then it should be declared with {}.  Since it is a plain struct.

Reply:
> So here's a question: for ~things~ members that have a meaningful default constructor, do we still put `{}` in the class? For POD types it's obviously required, but for `winrt::hstring` and `wil::unique_x` it's implicit and it's always initialized properly. Thoughts?

My historical approach has been that non-POD objects (classes) never get {} applied to them since the default constructor will be executed when the parent object is constructed because I trust the constructor of the class to initialize it.  I then applied {} to all PODs and structs because equally because I didn't want to have to consider whether they a constructor defined.

I think the guidance the ISO C++ Guidelines provide is simplest here:

[C.48: Prefer in-class initializers to member initializers in constructors for constant initializers](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers)

In short, if you look at the example there, the guidance seems to suggest that all members that should have an initial, constant value should have an in-class initializer and that the constructors for the class should then use member initializers in the constructors in member definition order.

Unfortunately, I don't think there's any "right, obvious answer" here.  So my personal suggestion would be to look at the relevant guideline, pick whatever interpretation is simplest to apply and go with that.  As long as it's consistent, I think everyone will be happy.

Reply:
> For enums and enum class we require that you provide a default enum value. (enums with {} have the potential to be completely wrong)

Personally, I have a similar, but slightly different approach.

enum or enum class members are always declared with the initial enum value (0) being an 'Unknown' or 'Invalid' value.  For classes and structs that have those enum members where I expect a value to be provided upon construction, the in-class member initializers have {}, which makes it easy to spot objects that were not constructed correctly since no valid struct/class should ever have those values for the enums.

Reply:
Interesting.  I have to run but I might ping you later and ask a few questions around this.  Always looking for better guidelines.   Thanks!

Reply:
I ended up choosing to use `{}` for POD and POD-likes, and standard default construction for non-POD things.